src/wallet/readable.go

```diff
@@ -63,7 +78,7 @@ func (res ReadableEntries) toWalletEntries(isEncrypted bool) ([]Entry, error) {
 		}
 
 		// Verify the wallet if it's not encrypted
-		if !isEncrypted {
+		if !isEncrypted && re.Secret != "" {
```

Comment:
The ReadableWallet that is decoded from http response, does not have secrets filed even it's unencrypted. And when call `ToWallet` method, it will fail here.